exercises/Toluwalase_Agoro/ex02_oo_basics/src/main.cpp

```diff
@@ -0,0 +1,57 @@
+#include "Shapes.h"
+
+int main(int, char **)
+{
+  Square smallSquare{4};
+  Square largeSquare{ 6 };
+  Square largerSquare{ 8 };
+  Square largestSquare{ 10 };
+ // std::cout << largeSquare.calculateArea() << "   " << largeSquare.calculatePerimeter();
+  Triangle smallTriangle{ 3,4 };
+  Triangle largeTriangle{ 6,8 };
+  Triangle largestTriangle{ 10,12 };
+
+  Rectangle smallRectangle{ 2,3 };
+  Rectangle largeRectangle{ 4,5 };
+  Rectangle largerRectangle{ 6,7 };
+  Rectangle largestRectangle{ 8,10 };
+
+  std::vector < BaseShape*> allShapes;
```

Comment:
This is an interesting approach - technically not wrong but have you considered using smart_pointers here rather than raw pointers. If you've not come across them yet I would recommend looking into them